storage/interceptors/notifications_interceptor.go

```diff
@@ -126,6 +127,10 @@ func (ni *NotificationsInterceptor) OnTxUpdate(h storage.InterceptUpdateOnTxFunc
 			}); err != nil {
 				return nil, err
 			}
+
```

Comment:
if the notification is not applied in the platform and we delete the credentials in smdb, the platform credentials will now work once again because of the fallback behaviour. is this ok ? More generally can we revise when the platform credentials should be usable and add tests for the cases?

I think platform credentials should be written as old_credentials/new_credentials initially (probably with migration?) and once they fall out due to rotations, fall backing to them should no longer work